src/main/resources/static/configurator.html

```diff
@@ -1,166 +1,266 @@
+                                    }
+                                    console.log(error);
+                                }
+                        );
+                    };
+
+                    this.addDevice = function (name, type, onUrl, offUrl) {
+                        this.state.error = "";
+                        return $http.post(this.state.base, {
+                            name: name,
+                            deviceType: type,
+                            onUrl: onUrl,
+                            offUrl: offUrl
+                        }).then(
+                                function (response) {
+                                    self.viewDevices();
+                                },
+                                function (error) {
+                                    if (error.data) {
+                                        self.state.error = error.data.message;
+                                    }
+                                    console.log(error);
+                                }
+                        );
+                    };
+
+                    this.deleteDevice = function (id) {
+                        this.state.error = "";
+                        return $http.delete(this.state.base + "/" + id).then(
+                                function (response) {
+                                    self.viewDevices();
+                                },
+                                function (error) {
+                                    if (error.data) {
+                                        self.state.error = error.data.message;
+                                    }
+                                    console.log(error);
+                                }
+                        );
+                    };
+                }])
+
+                .controller('ViewingController', ["$scope", "bridgeService", function ($scope, bridgeService) {
+                    bridgeService.viewDevices();
+                    $scope.bridge = bridgeService.state;
+                    $scope.deleteDevice = function (device) {
+                        bridgeService.deleteDevice(device.id);
+                    };
+                    $scope.testUrl = function (url) {
+                        window.open(url, "_blank");
+                    };
+                    $scope.setBridgeUrl = function (url) {
+                        bridgeService.state.base = url;
+                        bridgeService.viewDevices();
+                    }
+                }])
+
+                .controller('AddingController', ["$scope", "bridgeService", function ($scope, bridgeService) {
+
+                    $scope.bridge = bridgeService.state;
+                    $scope.device = {name: "", type: "switch", onUrl: "", offUrl: ""};
+                    $scope.vera = {base: "http://192.168.1.144:3480", id: 1};

```

Comment:
Yeah that was in the original code, unlike the bridgeURL we can't really make a smart guess on where Vera is. I do like that this shows the port Vera listens on, and that no slash or path is needed after this. Maybe this could be moved to the placeholder text though.